src/test/scenarios/attestation/output/src/attestation/azext_attestation/tests/__init__.py

```diff
@@ -34,16 +37,34 @@ def get_func_to_call():
         func_to_call = func
         try:
             func_to_call = import_manual_function(func)
+            print("Found manual override for {}(...)".format(func.__name__))
         except (ImportError, AttributeError):
             pass
         return func_to_call
 
     def wrapper(*args, **kwargs):
         func_to_call = get_func_to_call()
         print("running {}()...".format(func.__name__))
-        return func_to_call(*args, **kwargs)
+        try:
+            return func_to_call(*args, **kwargs)
+        except BaseException as e:
+            print("--------------------------------------")
+            print("step exception: ", e)
+            print("--------------------------------------", file=sys.stderr)
+            print("step exception in {}: {}".format(func.__name__, e), file=sys.stderr)
+            traceback.print_exc()
+            exceptions.append((func.__name__, sys.exc_info()))
 
     if inspect.isclass(func):
         return get_func_to_call()
     else:
         return wrapper
+
+
+def raise_if():
+    if exceptions:
+        if len(exceptions) <= 1:
+            raise exceptions[0][1][1]
+        message = "{}\nFollowed with exceptions in other steps:\n".format(str(exceptions[0][1][1]))
+        message += "\n".join(["{}: {}".format(h[0], h[1][1]) for h in exceptions[1:]])
+        raise exceptions[0][1][0](message).with_traceback(exceptions[0][1][2])

```

Comment:
Is exceptions a three demensions array? or one demension but with dictionary type element ?

Reply:
refer to Line 66:
~~~
output.push('            exceptions.append((func.__name__, sys.exc_info()))');
~~~

The exceptions is an array of (step_name, exception_info), where step_name is a string,  and exception_info is a 3 elements tuple: 
exception_info -> (exception_type, exception_instance, stack_info)

Reply:
okay. Is exception_instance a string type? 
`message = "{}\nFollowed with exceptions in other steps:\n".format(str(exceptions[0][1][1]))`

Reply:
No, it should be some subclass of BaseException

Reply:
Then are you sure this line work 
`message = "{}\nFollowed with exceptions in other steps:\n".format(str(exceptions[0][1][1]))`